aten/src/ATen/native/TensorTransformations.cpp

```diff
@@ -142,4 +142,67 @@ Tensor flipud(const Tensor& self) {
   return self.flip({0});
 }
 
+Tensor atleast_1d(const Tensor& self) {
+  switch (self.dim()) {
+    case 0:
+      return self.reshape({1});
+    default:
+      return self;
+  }
+}
+
+std::vector<Tensor> atleast_1d(TensorList tensors) {
+  std::vector<Tensor> result(tensors.size());
+  auto transform_lambda = [](const Tensor& input) -> Tensor {
+    return at::native::atleast_1d(input);
+  };
+  std::transform(tensors.begin(), tensors.end(), result.begin(), transform_lambda);
+  return result;
+}
+
+Tensor atleast_2d(const Tensor& self) {
+  switch (self.dim()) {
+    case 0:
+      return self.reshape({1, 1});
+    case 1: {
+      return self.unsqueeze(0);
+    }
+    default:
+      return self;
+  }
+}
+
+std::vector<Tensor> atleast_2d(TensorList tensors) {
+  std::vector<Tensor> result(tensors.size());
+  auto transform_lambda = [](const Tensor& input) -> Tensor {
+    return at::native::atleast_2d(input);
+  };
+  std::transform(tensors.begin(), tensors.end(), result.begin(), transform_lambda);
```

Comment:
Same comment about using constant iterators for the tensor list

Reply:
Done.